hardhat-project/contracts/SVG.sol

```diff
@@ -276,23 +270,172 @@ library SVG {
         }
     }
 
-    function _encodeColorTable(bytes memory data)
+    function _encodeAnimationFrames(uint8 framesCount, bytes memory data)
+        internal
+        pure
+        returns (bytes memory defs)
+    {
+        // Load hex chars into memory
+        bytes memory hexChars = HEX_CHARS;
+
+        // Output length
+        //
+        // 43 is the number of bytes used by the <g> container element:
+        //
+        // - 20 bytes - header
+        // -  1 byte  - id (a single hex char is enought to encode at most 16
+        //              different frames)
+        // -  6 bytes - header closing
+        // - 16 bytes - footer
+        uint256 defLength = RECT_SIZE * IMAGE_WIDTH * IMAGE_HEIGHT + 43;
+
+        // Allocate a new bytes variable
+        defs = new bytes(defLength * framesCount);
+
+        // Offset for reading frame data
+        // Jump over frames count (1) and packed fields (1)
+        uint256 frameDataOffset = 2;
+        // Offset for storing data in the defs bytes
+        uint256 defsOffset = 0;
+
+        for (uint256 f = 0; f < framesCount; f++) {
+            bytes memory bitmap = _encodeBitmap(data, frameDataOffset);
+
+            assembly {
+                /**
+                 * Returns two bytes that represent the input byte `b` encoded as a
+                 * hexadecimal string.
+                 *
+                 * `hexTable` is the pointer to the hexadecimal chars table (0-F).
+                 *
+                 * Examples:
+                 *
+                 * - `byteToHex(255, hexCharsPtr)` -> h = 'F', l = 'F'
+                 * - `byteToHex(128, hexCharsPtr)` -> h = '8', l = '0'
+                 * - `byteToHex(15,  hexCharsPtr)` -> h = '0', l = 'F'
+                 */
+                function byteToHex(b, hexTable) -> h, l {
+                    h := mload(add(hexTable, shr(4, and(b, 0xf0))))
+                    l := mload(add(hexTable, and(b, 0xf)))
+                }
+
+                /**
+                 * Same as: let h, _ := byteToHex(b, hexTable)
+                 */
+                function byteToHexH(b, hexTable) -> h {
+                    h := mload(add(hexTable, shr(4, and(b, 0xf0))))
+                }
+
+                /**
+                 * Same as: let _, l := byteToHex(b, hexTable)
+                 */
+                function byteToHexL(b, hexTable) -> l {
+                    l := mload(add(hexTable, and(b, 0xf)))
+                }
```

Comment:
I implemented these two functions because parser used by solhint doesn't support multi-assignments in Yul.

See https://github.com/solidity-parser/antlr/pull/11 and https://github.com/solidity-parser/parser/issues/60